sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/KafkaProperties.java

```diff
@@ -18,6 +21,33 @@ private KafkaProperties() {
     //
   }
 
+  private static Properties configureSecurity(Security security) {
+    KafkaProperties props = new KafkaProperties();
+
+    if (security == null) {
+      return props;
+    }
+
+    if (security.getProtocol() != null) {
+      props.put("security.protocol", security.getProtocol().name());
+    }
+
+    if (security.getPassword() != null && security.getUser() != null) {
+      props.put("sasl.mechanism", security.getSaslMechanism());
+      props.put(
+          "sasl.jaas.config",
+          getLoginModule(security.getSaslMechanism())
+              .getName()
+              .concat(" required username='")
+              .concat(security.getUser())
```

Comment:
NIT: Same as before, but don't we have to escape username and password here?

Reply:
I'm not sure what you mean with "escape" here.

Reply:
I'm not sure what happens if the password (or less likely the username) contains `'`. So it may be required to convert a password like `my'secret` into `my\'secret` or anything else how strings have to be escaped here. But we can postpone that discussion and wait until a problem is reported.